openfisca_aotearoa/variables/live_organ_donor_compensation.py

```diff
@@ -0,0 +1,131 @@
+        earnings_excluding_compensation_period = earnings_period - compensation_period
+
+        self_employed_earnings_in_most_recent_tax_year = persons('sum_of_self_employed_earnings_in_most_recently_completed_tax_year', period)
+        most_recent_tax_year_period = persons('number_of_weeks_in_most_recently_completed_tax_year', period)
+
+        employee_weekly_earnings = earnings_excluding_compensation_amount / earnings_excluding_compensation_period
+        self_employed_weekly_earnings = self_employed_earnings_in_most_recent_tax_year / most_recent_tax_year_period
+        
+        return round(employee_weekly_earnings + self_employed_weekly_earnings, 2)
+
+class kiwisaver_employee_deduction(Variable):
+    value_type = float
+    entity = Person
+    definition_period = YEAR # TODO - determine whether we need to get WEEK to work
+    label = u"The amount deducted as a kiwisaver contribution"
+
+    def formula(persons, period, parameters):
+        earnings_amount = persons('sum_of_earnings_in_last_52_weeks', period) 
+        compensation_amount = persons('sum_of_earnings_during_compensation_period_in_last_52_weeks', period)
+        earnings_excluding_compensation_amount = earnings_amount - compensation_amount
+
+        earnings_period = persons('earnings_period_in_weeks', period)
+        compensation_period = persons('compensation_period_in_weeks', period)
+        earnings_excluding_compensation_period = earnings_period - compensation_period
+
+        self_employed_earnings_in_most_recent_tax_year = persons('sum_of_self_employed_earnings_in_most_recently_completed_tax_year', period)
+        most_recent_tax_year_period = persons('number_of_weeks_in_most_recently_completed_tax_year', period)
+
+        employee_weekly_earnings = earnings_excluding_compensation_amount / earnings_excluding_compensation_period
+        self_employed_weekly_earnings = self_employed_earnings_in_most_recent_tax_year / most_recent_tax_year_period
+        
+        weekly_compensation = round(employee_weekly_earnings + self_employed_weekly_earnings, 2)
+
+        kiwisaver_employee_deduction_percentage = persons('kiwisaver_employee_deduction_percentage', period)
+        return round(weekly_compensation * kiwisaver_employee_deduction_percentage / 100, 2)
+
+# class PayFrequency(Enum):
+#     weekly = u'Weekly'
+#     fortnightly = u'Fortnightly'
+#     four_weekly = u'Four Weekly'
+#     monthly = u'Monthly'
+
+# class pay_frequency(Variable):
+#     value_type = Enum
+#     possible_values = PayFrequency
+#     default_value = PayFrequency.weekly  # The default is mandatory
+#     entity = Person
+#     definition_period = YEAR
+#     label = u"The frequency at which a person is paid"
+
+# class salary_per_pay(Variable):
+#     value_type = float
+#     entity = Person
+#     definition_period = YEAR # TODO - determine whether we need to get WEEK to work
+#     label = u"The salary earned by a person before tax"
```

Comment:
Version control (Git) can help us recover code if we need it, but commented-out code usually becomes a problem over time as we cannot know if it is still up-to-date or not… Could we maybe remove these parts from the model? 🤔